src/Datadog.Trace/Sampling/RuleBasedSampler.cs

```diff
@@ -9,37 +8,30 @@ internal class RuleBasedSampler : ISampler
         private const ulong KnuthFactor = 1_111_111_111_111_111_111;
 
         private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<RuleBasedSampler>();
+        private static bool _tracingWithoutLimitsEnabled = false;
 
         private readonly IRateLimiter _limiter;
+        private readonly DefaultSamplingRule _defaultRule = new DefaultSamplingRule();
         private readonly List<ISamplingRule> _rules = new List<ISamplingRule>();
 
-        private Dictionary<string, float> _sampleRates = new Dictionary<string, float>();
-
         public RuleBasedSampler(IRateLimiter limiter)
         {
             _limiter = limiter ?? new RateLimiter(null);
+            RegisterRule(_defaultRule);
         }
 
-        public void SetDefaultSampleRates(IEnumerable<KeyValuePair<string, float>> sampleRates)
+        public static void OptInTracingWithoutLimits()
```

Comment:
Note that this enables TwL for all `RuleBasedSampler` instances, even through each `Tracer` instance can have different settings (some with custom rules and some without).